src/Microsoft.Framework.Configuration/ConfigurationSection.cs

```diff
@@ -19,6 +20,11 @@ public ConfigurationSection(IList<IConfigurationSource> sources, string key)
                 throw new ArgumentNullException(nameof(sources));
             }
 
+            if (prefix == null)
+            {
+                throw new ArgumentNullException(nameof(prefix));
+            }
+
             if (key == null)
```

Comment:
I don't think we should allow empty keys (even if empty prefixes are fine). Am I missing something again? :smile: